aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -794,8 +836,8 @@ static Tensor &std_var_out(Tensor &result, const Tensor &self, IntArrayRef dim,
               "std and var only supports CPU AND CUDA device type, got: ", self.device().type());
   TORCH_CHECK(self.layout() == Layout::Strided,
               "std and var only supports strided layout, got: ", self.layout());
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()) || at::isComplexType(self.scalar_type()),
-              "std and var only support floating-point dtypes");
+  TORCH_CHECK(!at::isIntegralType(result.scalar_type(), /*includeBool=*/true),
```

Comment:
are these equivalent?

Reply:
Excluding quantized type, yes.

Reply:
Want to switch to `at::isFloatingType ... || at::isComplexType ...` for consistency?

Reply:
sgtm.  I believe @nairbv is working on a sub-dtype system like NumPy has that would make this cleaner in the future.